idaes/core/base/property_set.py

```diff
@@ -40,25 +40,27 @@ def __init__(
         required: bool = False,
         valid_range=None,
     ):
-        super().__setattr__("_parent", parent)
-        super().__setattr__("_idx", idx)
-        super().__setattr__("_method", method)
-        super().__setattr__("_supported", supported)
-        super().__setattr__("_required", required)
-
-        super().__setattr__("_valid_range", None)
+        self._parent = parent
+        self._idx = idx
+        self._method = method
+        self._supported = supported
+        self._required = required
+        self._valid_range = None
         self._set_valid_range(
             valid_range
         )  # This method does some basic validation of input
-
+        self._lock_setattr = True
         # TODO: For future, this would be the place to store default scaling information, etc.
         # TODO: Could also define default bounds, nominal values, etc.
 
     def __setattr__(self, key, value):
-        raise TypeError("Property metadata does not support assignment.")
+        try:
+            assert self._lock_setattr == True
```

Comment:
Should this be an `if/raise`? At the moment you will get an `AssertionError` and the `TypeError` will never trigger.

Reply:
It could be `if hasattr(...) and self._lock_setattr == True:`.  I think this works right though since the tests didn't break. Since the lock `self._lock_setattr = True` is added at the end of `__init__`, the assert will pass and the TypeError will be raised, once you pass the `self._lock_setattr = True` line in `__init__`.  At least I'm pretty sure this is tested, that you can't set new attributes. Maybe the if is easier to understand though.

Reply:
It think that would be better, otherwise the user could get a bare `AssertionError`.